vllm/worker/hpu_model_runner.py

```diff
@@ -298,9 +298,19 @@ def _set_block_mapping(self, metadata, batch_size, device, dtype):
         mask = mask >= metadata.block_usage.unsqueeze(-1)
         attn_bias = (torch.zeros_like(mask, dtype=dtype).masked_fill_(
             mask, -math.inf))
-        block_mapping = torch.nn.functional.one_hot(
-            metadata.block_mapping.to(torch.long),
-            num_classes=batch_size).to(dtype)
+        if 'cpu' in str(device).lower():
```

Comment:
maybe you can use 'not is_hpu()' from vllm.utils?

Reply:
Hmm, I guess is_fake_hpu would be better here as we want this code to run with fake_hpu_mode and afaik is_hpu returns True when running in fake mode.